libraries/botframework-schema/src/teams/index.ts

```diff
@@ -1225,6 +1229,167 @@ export interface FileConsentCardResponse {
     uploadInfo?: FileUploadInfo;
 }
 
+/**
+ * @interface
+ * Current tab request context, i.e., the current theme.
+ */
+export interface TabContext {
+    /**
+     * @member {theme} [string] The current user's theme.
+     */
+    theme?: string;
+}
+
+/**
+ * @interface
+ * Current TabRequest entity context, or 'tabEntityId'.
+ *
+ */
+export interface TabEntityContext {
+    /**
+     * @member {string} [tabEntityId] The entity id of the tab.
+     */
+    tabEntityId?: string;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/fetch') request value payload.
+ *
+ */
+export interface TabRequest {
+    /**
+     * @member {TabEntityContext} [tabContext] The current tab entity request context.
+     */
+    tabContext?: TabEntityContext;
+    /**
+     * @member {TabContext} [context] The current user context, i.e., the current theme.
+     */
+    context?: TabContext;
+    /**
+     * @member {string} [state] The magic code for OAuth flow.
+     */
+    state?: string;
+}
+
+/**
+ * @interface
+ * Envelope for Card Tab Response Payload.
+ *
+ */
+export interface TabResponse {
+    /**
+     * @member {TabResponsePayload} [tab] The response to the tab/fetch message.
+     */
+    tab?: TabResponsePayload;
```

Comment:
@clearab is a TabResponse without a TabResponsePayload a valid scenario? The issue/spec (microsoft/botframework-sdk#6138) made it seem like this isn't the case. 

Making `TabResponse.tab` required will tighten up the API and protect developers from sending malformed TabResponses.

This will differ from C# because the TabResponse is a partial class in C#, whereas TabResponse is an interface here.

@mdrichardson I'm thinking we should preemptively tighten this in TS to help protect developers. If Teams adds more scenarios with different TabResponse members we can loosen the reins a bit.

@joshgummersall @EricDahlvang as FYI

Reply:
Oh yeah, I meant to add as a comment that I'm not sure which of these properties are required/optional. Input here would be much appreciated.

Reply:
A fully populated response is required for to the tab/fetch invoke.

Reply:
I plan to make the following properties required:

* [TabResponse.tab](https://github.com/microsoft/botbuilder-js/pull/3224/files/07ba1ea22cdb949e0230102bcdc6441871d7fed5#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1284) (required, per @clearab)
* [TabResponseCard.card](https://github.com/microsoft/botbuilder-js/pull/3224/files/07ba1ea22cdb949e0230102bcdc6441871d7fed5#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1296) (it's the only property so it makes sense to require it, if they're going to use it in TabResponseCards.cards.
* [TabResponseCards.cards](https://github.com/microsoft/botbuilder-js/pull/3224/files/07ba1ea22cdb949e0230102bcdc6441871d7fed5#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1308) (it's the only property so it makes sense to require it, if they're going to use it in TabResponsePayload.value.
* [TabSuggestedActions.actions](https://github.com/microsoft/botbuilder-js/pull/3224/files/07ba1ea22cdb949e0230102bcdc6441871d7fed5#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1377) (similar to above. It's the only one and it should be required if you specify TabresponsePayload.suggestedActions.

Reply:
@clearab @stevengum That sound about right?

Reply:
Yes, that seems right.

Reply:
lgtm

Reply:
Sounds good to me, thanks everyone.